File: CondoManager.Domain/Entities/AccessLog.cs
namespace CondoManager.Domain.Entities;

public class AccessLog
{
    public Guid Id { get; set; } = Guid.NewGuid();
    
    public Guid VisitorId { get; set; }
    public Visitor Visitor { get; set; }
    
    public Guid EmployeeId { get; set; }
    public Employee Employee { get; set; }
    
    public DateTime TimeOfEntry { get; set; }
    public DateTime? TimeOfExit { get; set; }
    
    public Guid CondominiumId { get; set; }
    public Condominium Condominium { get; set; }
}

File: frontend/src/types/accessLog.ts
export interface AccessLog {
    id: string
    visitorId: string
    employeeId: string
    timeOfEntry: string
    timeOfExit: string | null
    condominiumId: string
}

export interface CreateAccessLogDto {
    visitorId: string
    employeeId: string
    condominiumId: string
}